builtin/logical/pki/path_tidy.go

```diff
@@ -1396,35 +1454,49 @@ func (b *backend) doTidyAcme(ctx context.Context, req *logical.Request, logger h
 	defer b.acmeAccountLock.Unlock()
 
 	sc := b.makeStorageContext(ctx, req.Storage)
-	thumbprints, err := sc.Storage.List(ctx, acmeThumbprintPrefix)
-	if err != nil {
-		return err
-	}
-
-	b.tidyStatusLock.Lock()
-	b.tidyStatus.acmeAccountsCount = uint(len(thumbprints))
-	b.tidyStatusLock.Unlock()
+	var after string
+	var thumbprintsCount uint
 
-	for _, thumbprint := range thumbprints {
-		err := b.tidyAcmeAccountByThumbprint(b.acmeState, sc, thumbprint, config.SafetyBuffer, config.AcmeAccountSafetyBuffer)
+	for {
+		thumbprints, err := sc.Storage.ListPage(ctx, acmeThumbprintPrefix, after, config.PageSize)
 		if err != nil {
-			logger.Warn("error tidying account %v: %v", thumbprint, err.Error())
+			return err
 		}
 
-		// Check for cancel before continuing.
-		if atomic.CompareAndSwapUint32(b.tidyCancelCAS, 1, 0) {
-			return tidyCancelledError
+		// If no thumbprints are returned, we've reached the end of the list
```

Comment:
While you're here, do you mind adding status messages for this and EAB as well?

Reply:
Alrighty, done! I'll push the changes in a minute after I double check all the changes :)